### DISCUSSION TITLE
How can I get the app thinning sizes for all devices for a version

### DISCUSSION THREAD

**Discussion by ZhangsongLee** [fastlane/fastlane#18302] [Q&A]
i want to get the thinning size for all the devices,
I used tunes_all_builds_for_train  in Spaceship::Tunes::Application before
```
app = Spaceship::Tunes::Application.find(XXXX)
app.tunes_all_builds_for_train(train:X.X.X)
builds.each { |build|
        details = build.details
        result << {'size':details.sizes_in_bytes.to_json,'build_id':details.cf_bundle_version,'version':train,'upload_date':details.upload_date}
    }
```
now I want to use Spaceship::ConnectAPI::App, but I did not find a function that could get the sizes

**Comment by dinsen** [comment]
Hi @CNOFNe 
Have you changed to use the Api Key also?
If thats the case I don't think the App Store Connect Api returns that information.

You can take a look in the documentation here:
https://developer.apple.com/documentation/appstoreconnectapi

**Reply by ZhangsongLee** [reply]
yes，i changed to use the Api Key.
i don't find App Store Connect Api returns that information. :(
